lib/main.js

```diff
@@ -23,6 +24,7 @@ const { isNonEmpty } = require('./utils')
  * @property {boolean|undefined} [noStyle] - when set, skips style rules
  * @property {boolean|undefined} [semi] - when set, enforces rather than forbids semicolons
  * @property {boolean|undefined} [ts] - when set, enables TypeScript checks
+ * @property {boolean|import('./react').ReactConfigName|import('./react').ReactConfig|undefined} [react] = when set, enables React checks, using selected config or 'react/recommended', see {@link https://www.npmjs.com/package/eslint-plugin-react|React plugin Docs}
```

Comment:
What would the drawbacks of enabling it unconditionally without adding a tunable? 

The eslint configuration is only triggered on JSX/TSX files.  

The ruleset to use should also not be opened to user configuration. The goal is to give a consistent neostandard behavior by default for JSX/TSX: rules matching it should be extracted and enabled. If user what to change the default behavior, he still can do it in the ESLint configuration.

Reply:
The reason I made it configurable, and perhaps that was a bad decision-- against the idea of this project-- was that react plugin has 3 different configurations (especially `jsx-runtime` to support some sort of special JSX transform from react17). I always used `recommended`, but I wasn't sure what would other users need/prefer. 
Now, you can extend `neostandard` rules easily, like this:
```
export default [
  ...neostandard(),
  jsdoc.configs['flat/recommended-typescript-flavor'],
]
```
but that would not work for plugins, like this one. I somehow had to pass "the options" to `reactify` function, to build a proper config. That could be due to the lack of knowledge of how to set up `neostandard`, but without touching the code (this PR), I could not find a way how to set it all up to work with TSX, only JSX.
So if I bake everything in and use, for example `recommended` config, `neostandard` users won't be able to use `jsx-runtime` or `all`, unless they use their own ESLint configuration. But can they use their own ESLint configuration and have `neostandard` rules as well?
Also, I wasn't sure why TypeScript is enabled by default and we have to use `ts: true` switch. That's why I decided to add another `react` switch and did not enable React support by default.
If someone can decide which ruleset from react plugin I should use, I'm happy to change it and make it non-configurable, always using that ruleset. I just feel like I cannot make that decision.

Reply:
Understood. I think starting with `recommended` is a good starting point. `all` is a bit too opinionated and I'm not sure about all those rules. I can also change it not to use any of the rules from the plugin, just use the plugin to support JSX and start linting the files using only rules from `neostandard`. Thoughts? I did study a bit more the "react settings" and I believe I can remove the object from `react` option and use the default values. So it can be just a `string` to enable (add) plugin rules. By default it would be enabled, as mentioned only with `neostandard` rules and with string value it would add the rules from the plugin.

Reply:
OK. I now understand what better, what I'm doing 🤦 (I'm not a React expert so...)...
There is a conflict in some rules. For example, neostandard rule `no-unused-vars` conflicts with `react/jsx-uses-react`. Because `neostandard` says you should not import `React` and do not use it, but old React needed it for JSX to work. That's why there is `jsx-runtime` for React 18+ where this is no longer necessary. So if I do not import the rules (at least recommended), I will always have this error (unused React const).
I will have to detect React version and use "recommended" or "all" + jsx-runtime (if it is React 18+)

Reply:
Sorry for all this noise... I have updated `reactify` to always enable JSX support with `recommended` rule set. Ruleset can be changed to `all` and that is the only option `neostandard` user has at this moment. I think that `neostandard` should either adapt one of the rule sets (but `all` is in my opinion too opinionated) or define it's own and accept only some rules from `all`. But, as I said many times, I'm not sure I can decide which rules should be "adopted" and which skipped. I will try again to use just the plugin with no rules, just to add simple JSX support. I think I now understand better how to do this.

Reply:
I would have to pick some of the rules from: https://github.com/jsx-eslint/eslint-plugin-react/blob/8cf47a8ac2242ee00ea36eac4b6ae51956ba4411/index.js#L165-L179
Important ones for example are: `react/jsx-uses-react` and/or `react/react-in-jsx-scope` because linter does (not) complain about unused vars.

Reply:
I have checked `recommended` rule set and I believe it should be ok to start with this, remove the `react` option and do not let the user choose anything else, except `requireReactImport` for old React.